src/kernel/kshell.c

```diff
@@ -184,6 +184,7 @@ int try_exec(int argc, char* argv[])
 
   char* buf = malloc((stat.size + 1) * sizeof(char));
   vfs_read(inode, buf, stat.size, 0);
+  buf[stat.size] = '\0';
```

Comment:
Is this still needed?

Reply:
It is needed to null-terminate the string since tar no longer terminates what it reads.